model/task_queue_service.go

```diff
@@ -60,16 +60,38 @@ func (s *taskDispatchService) FindNextTask(distroID string, spec TaskSpec) (*Tas
 }
 
 func (s *taskDispatchService) RefreshFindNextTask(distroID string, spec TaskSpec) (*TaskQueueItem, error) {
+	start := time.Now()
 	distroDispatchService, err := s.ensureQueue(distroID)
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
+	grip.Debug(message.Fields{
+		"step":          "RefreshFindNextTask",
+		"op":            "ensureQueue",
+		"duration":      time.Since(start),
+		"duration_secs": time.Since(start).Seconds(),
+	})
+	start = time.Now()
 
 	if err := distroDispatchService.Refresh(); err != nil {
 		return nil, errors.WithStack(err)
 	}
+	grip.Debug(message.Fields{
+		"step":          "RefreshFindNextTask",
+		"op":            "Refresh",
+		"duration":      time.Since(start),
+		"duration_secs": time.Since(start).Seconds(),
+	})
+	start = time.Now()
 
-	return distroDispatchService.FindNextTask(spec), nil
+	item := distroDispatchService.FindNextTask(spec)
+	grip.Debug(message.Fields{
+		"step":          "RefreshFindNextTask",
```

Comment:
It might be worth logging something about the item here. Though what I think would be most interesting would actually be how deep item is in the queue, but you don't have access to that here